pmoves/services/token-stub/app.py

```
@@ -0,0 +1,211 @@
+    try:
+        os.makedirs(os.path.dirname(EVIDENCE_LOG), exist_ok=True)
+        with open(EVIDENCE_LOG, "a") as f:
+            f.write(json.dumps(record, sort_keys=True) + "\n")
+        log.debug("evidence written for work_id=%s", record.get("work_id"))
+    except Exception as exc:
+        log.warning("write_evidence failed: %s", exc)
+
+
+def build_rollback_plan(work_id: str) -> list[str]:
+    return [
+        f"DELETE FROM work_attestations WHERE work_id='{work_id}';",
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)